### PR TITLE
[libc] Add proxy header for the stack_t type

### PR DESCRIPTION
**Pull request by wldfngrs**
added proxy header for the stack_t type and modified the corresponding CMakeLists.txt files

### CODE REVIEW

**Comment by wldfngrs** [comment]
@lntue, please review!

File: libc/hdr/types/stack_t.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
+//===-- Definition of macros from stack_t.h -------------------------------===//
```
Comment (lntue) [inline libc/hdr/types/stack_t.h:1]: Nit: `Definition of stack_t`

File: libc/src/signal/sigaltstack.h [hunk 2]
```diff
@@ -9,8 +9,8 @@
 #ifndef LLVM_LIBC_SRC_SIGNAL_SIGALTSTACK_H
 #define LLVM_LIBC_SRC_SIGNAL_SIGALTSTACK_H
 
+#include "hdr/types/stack_t.h"
```
Comment (lntue) [inline libc/src/signal/sigaltstack.h:12]: Update dependency of `src/signal/linux/signalstack`

Comment (wldfngrs) [reply libc/src/signal/sigaltstack.h:12]: Build fails when I update the dependency with `libc.hdr.types.stack_t` in the `CMakeLists.txt` file in the `linux` directory

File: libc/hdr/types/CMakeLists.txt [hunk 3]
```diff
@@ -181,3 +181,12 @@ add_proxy_header_library(
     libc.include.signal
 )
 
+add_proxy_header_library(
+  stack_t
+  HDRS
+    stack_t
```
Comment (lntue) [inline libc/hdr/types/CMakeLists.txt:187]: `stack_t.h`

**Review by lntue — APPROVED** [review]

**Pull request merged**